.evergreen/compile.sh

```diff
@@ -156,6 +156,10 @@ linux*)
   else
     cxx_flags+=("-Wno-unused-parameter") # TODO: remove once C driver is upgraded to include fix of CDRIVER-5673.
   fi
+
+  if [[ "${distro_id:?}" == debian12* ]]; then
+    cxx_flags+=("-Wno-error=restrict")
```

Comment:
Suggest commenting describing why this is needed.
```suggestion
    # Disable `restrict` warning on GCC 12 due to  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651 and https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105329
    cxx_flags+=("-Wno-error=restrict")
```